src/components/top_bar/top_bar.ts

```diff
@@ -254,30 +164,32 @@ css/* scss */ `
 `;
 export class TopBar extends Component<Props, SpreadsheetChildEnv> {
   static template = "o-spreadsheet-TopBar";
-  static components = { ColorPickerWidget, Menu, TopBarComposer, FontSizeEditor };
-  commonFormats = FORMATS;
-  customFormats = CUSTOM_FORMATS;
-  currentFormatName = "automatic";
-
   get dropdownStyle() {
     return `max-height:${this.props.dropdownMaxHeight}px`;
   }
+  static components = {
+    ColorPickerWidget,
+    ColorPicker,
+    Menu,
+    TopBarComposer,
+    FontSizeEditor,
+    MenuItemButton,
+  };
 
-  style: Style = {};
   state: State = useState({
     menuState: { isOpen: false, position: null, menuItems: [] },
     activeTool: "",
+    fillColor: "#ffffff",
+    textColor: "#000000",
   });
   isSelectingMenu = false;
   openedEl: HTMLElement | null = null;
-  inMerge = false;
-  cannotMerge = false;
-  undoTool = false;
-  redoTool = false;
-  paintFormatTool = false;
-  fillColor: string = "#ffffff";
-  textColor: string = "#000000";
   menus: MenuItem[] = [];
+  editMenuItems = editMenuItems;
+  formatMenuItems = formatMenuItems;
+  viewMenuItems = viewMenuItems;
+  formatNumberMenuItem = createMenuItem(formatNumberMenuItemSpec);
+  isFeatureContextMenu = false;
```

Comment:
what's this flag ?

Reply:
Not well named.
Here, we need to differentiate between a menu that opens in the toolbar (eg the format button) and a menu that opens in the topbar (file/edit/view/insert/...).
The difference is that the topbar menu must open automatically with the mouse hover (when at least one of these is open) while this is not the case with the toolbar menus

Reply:
isntToolbarMenu will be better